smack-extensions/src/main/java/org/jivesoftware/smackx/vcardtemp/provider/VCardProvider.java

```diff
@@ -18,339 +18,335 @@
+        EXTADR,
+        STREET,
+        LOCALITY,
+        REGION,
+        PCODE,
+        CTRY,
+        FF
+    }
 
-    static {
-        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
-        try {
-            documentBuilderFactory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
-        } catch (ParserConfigurationException e) {
-            LOGGER.finer("Could not disallow doctype decl: " + e.getMessage());
-            // If we can't disable DTDs, then at least try the following
-            // Xerces 1 - http://xerces.apache.org/xerces-j/features.html#external-general-entities
-            // Xerces 2 - http://xerces.apache.org/xerces2-j/features.html#external-general-entities
-            try {
-                documentBuilderFactory.setFeature( "http://xml.org/sax/features/external-general-entities", false);
-                documentBuilderFactory.setFeature( "http://xml.org/sax/features/external-parameter-entities", false);
-            } catch (ParserConfigurationException e1) {
-                LOGGER.finer("Could not disallow external entities for xerces parser: " + e1.getMessage());
-            }
-        }
-        // Android throws an UnsupportedOperationException when calling setXIncludeAware() and for some dumb reason also
-        // when calling isXIncludeWare(), while it defaults according to the docs to 'false'.
-        boolean isXIncludeAware;
-        try {
-            isXIncludeAware = documentBuilderFactory.isXIncludeAware();
-        } catch (UnsupportedOperationException e) {
-            // Assume we are on Android where isXIncludeAware defaults to 'false'
-            isXIncludeAware = false;
-        }
-        if (isXIncludeAware) {
-            documentBuilderFactory.setXIncludeAware(false);
-        }
-        documentBuilderFactory.setExpandEntityReferences(false);
+    private enum Tel {
+        VOICE,
+        FAX,
+        PAGER,
+        MSG,
+        CELL,
+        VIDEO,
+        BBS,
+        MODEM,
+        ISDN,
+        PCS,
+        PREF
+    }
 
-        // Harden the parser even further
-        documentBuilderFactory.setIgnoringComments(true);
-        documentBuilderFactory.setCoalescing(false);
-        documentBuilderFactory.setValidating(false);
+    @Override
+    public VCard parse(XmlPullParser parser, int initialDepth) throws XmlPullParserException, IOException,
+                    SmackException {
+
+        VCard vCard = new VCard();
 
         try {
-            documentBuilderFactory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
-        } catch (ParserConfigurationException e) {
-            LOGGER.info("Could not enable secure processing parsing feature: " + e.getMessage());
-        }
 
-        DOCUMENT_BUILDER_FACTORY = documentBuilderFactory;
-    }
+            boolean done = false;
+            boolean lastEventWasStart = false;
+            int depth = 0;
+            String name = "";
```

Comment:
Those values should usually be `null` initialized. Otherwise you can't tell the difference between a missing attribute value/text or if it's set to the empty string.